File: flow-sample-helloworld/src/androidTest/java/flow/sample/helloworld/HelloTest.java
package flow.sample.helloworld;

import org.junit.Rule;
import org.junit.Test;

import androidx.test.rule.ActivityTestRule;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.matcher.ViewMatchers.withText;

public class HelloTest {
  @Rule public ActivityTestRule rule = new ActivityTestRule<>(HelloWorldActivity.class);

  @Test public void hello() {
    onView(withText("Hello, World!")).perform(click());
  }
}

File: flow-sample-helloworld/src/androidTest/java/flow/sample/helloworld/HelloTest.kt
package flow.sample.helloworld

import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.action.ViewActions.click
import androidx.test.espresso.matcher.ViewMatchers.withText
import androidx.test.ext.junit.rules.activityScenarioRule
import org.junit.Rule
import org.junit.Test

class HelloTest {
  @Rule
  @JvmField
  val rule = activityScenarioRule<HelloWorldActivity>()

  @Test
  fun hello() {
    onView(withText("Hello, World!")).perform(click())
  }
}